Automattic/vip-go-mu-plugins
# Add wp-parsely loader information to SDS

## Description

This adds wp-parsely plugin-related information to the site details index, with the helper of a new `Parsely_Loader_Info` class that is introduced to help keep track of how (and if) the plugin is being loaded.

Additionally:

- Avoid fatal errors when activating a self-installed version of the wp-parsely plugin whilst the mu-plugins version was already active.
- Cleanup loader logic, so that we can more effectively take into account the `wpvip_parsely_load_mu` filter & `_wpvip_parsely_mu` option.
- Move `maybe_load_plugin()` logic to the later `plugins_loaded` hook priority 1, rather than `muplugins_loaded`. This allows us to safely ensure a self-installed version of the plugin is not being loaded.

## Changelog Description
Information about how a site uses the parse.ly plugin has been added to the Site Details Index. The bundled version of wp-parsely plugin now loads on the `plugins_loaded` hook rather than on `muplugins_loaded` when enabled.

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to test

Try out the various states, ensure all is loading correctly & reporting the correct loader information.

- Parsely not enabled.
- Parsely enabled by the option.
- Parsely enabled by the filter.
- Parsely enabled via a self-installed version.
- Parsely enabled by the option but disabled by the filter.
